joladnijo/models.py

```diff
@@ -0,0 +1,171 @@
+
+    def assets_fulfilled(self):
+        return self.assetrequest_set.fulfilled()
+
+    def assets_overloaded(self):
+        return self.assetrequest_set.overloaded()
+
+    def _changes_of(self, history, field = 'self', id = None):
+        changes = []
+        field_prefix = None
+        if field != 'self':
+            field_prefix = field
+            if id is not None:
+                field_prefix += '.%s' % id
+        for record_pair in pair_iterable_for_delta_changes(history.order_by('history_date').iterator()):
+            old_record, new_record = record_pair
+            delta = new_record.diff_against(old_record)
+            for change in delta.changes:
+                changes.append({
+                    # TODO: legyen külön mezőben a field és id, vagy jó így?
+                    'changed': change.field if field_prefix is None else '%s.%s' % (field_prefix, change.field),
+                    'from': change.old,
+                    'to': change.new,
+                    'date_time': new_record.history_date,
+                })
+        return changes
+
+    def feed(self):
+        # TODO: ezt talán jó lenne valahogy gyorsítótárazni
+        # TODO: mi kell pontosan a feedbe: elég self és assetrequest?
+        changes = self._changes_of(self.history.all())
+        for request in self.assetrequest_set.all():
+            changes += self._changes_of(request.history.all(), 'assetrequest', request.id)
+        changes += self._changes_of(self.organization.history.all(), 'organization')
+        if self.contact is not None:
+            changes += self._changes_of(self.contact.history.all(), 'contact')
+        changes.sort(key=lambda c: c['date_time'], reverse=True)
+        return changes
+
+    def __str__(self) -> str:
+        return '%s - %s (%s)' % (self.organization.name, self.name, self.city)
+
+
+class Contact(BaseModel, NoteableModel):
+    name = models.CharField(max_length=255, blank=False)
+    email = models.EmailField(max_length=255, blank=False)
+    phone = models.CharField(max_length=20, blank=True)
+    facebook = models.URLField(max_length=255, blank=True)
+    url = models.URLField(max_length=255, blank=True)
+    organization = models.OneToOneField(Organization, on_delete=models.CASCADE, blank=True, null=True)
+    aid_center = models.OneToOneField(AidCenter, on_delete=models.CASCADE, blank=True, null=True)
+
+    def __str__(self) -> str:
+        if self.phone == '':
+            return '%s (%s)' % (self.name, self.email)
+        return '%s (%s, %s)' % (self.name, self.phone, self.email)
+
+
+class AssetCategory(BaseModel):
+    name = models.CharField(max_length=255, blank=False, unique=True)
+    parent = models.ForeignKey('self', blank=True, null=True, on_delete=models.SET_NULL)
+    icon = models.CharField(max_length=50, blank=True)
+
+    class Meta:
+        verbose_name_plural = 'Asset categories'
+
+    def __str__(self) -> str:
+        return self.name
```

Comment:
Javaslat:

```python
def __str__(self) -> str:
  if self.parent in not None:
    return "%s (%s)" % (self.name, self.parent.name) # vagy akár egy `parent_name` oszlopban cache-elni
  return self.name
```

Reply:
Kicsit kapcsolódik - a jelenlegi koncepcióban ugye csak 2 szintnyi kategória lehet? Ha igen, ezt is meg kell még csinálni itt, jelenleg akármennyi megengedett.